lib/omnibus/packagers/pkgng.rb

```diff
@@ -0,0 +1,574 @@
+      end
+    end
+    expose :runtime_dependencies
+
+    #
+    # Set or return the options for this package.
+    #
+    # @example
+    #   options {
+    #     "OPT1" => "off"
+    #   }
+    #
+    # @param [Hash] val
+    #   the options for this package
+    #
+    # @return [Hash]
+    #   the options for this package
+    #
+    def options(val = NULL)
+      if null?(val)
+        @options || {}
+      else
+        unless val.is_a?(Hash)
+          raise InvalidValue.new(:options, "be a Hash")
+        end
+
+        @options = val
+      end
+    end
+    expose :options
+
+    #
+    # @!endgroup
+    # --------------------------------------------------
+
+    #
+    # The name of the package to create.
+    #
+    # @return [String]
+    #
+    def package_name
+      "#{safe_base_package_name}-#{pkgng_version}.txz"
+    end
+
+    #
+    # Read all scripts in {Project#package_scripts_path}
+    #
+    # @return [void]
+    #
+    def package_scripts
+      SCRIPT_MAP.reduce({}) do |scripts, (source, destination)|
+        source_path = File.join(project.package_scripts_path, source.to_s)
+
+        if File.file?(source_path)
+          log.debug(log_key) { "Adding script `#{source}'" }
+          scripts[destination] = File.read(source_path)
+        end
+
+        scripts
+      end
+    end
+
+    def compact_manifest
+      {
+        name: safe_base_package_name,
+        version: pkgng_version,
+        origin: origin,
+        comment: comment,
+        arch: safe_architecture,
+        www: project.homepage,
+        maintainer: safe_maintainer,
+        prefix: prefix,
+        licenses: licenses,
+        flatsize: package_size,
+        users: users,
+        groups: groups,
+        options: options,
+        desc: project.description,
+        categories: categories,
+        deps: runtime_dependencies,
+      }
+    end
+
+    def manifest
+      {
+        files: package_files,
+        directories: {},
+        scripts: package_scripts,
+      }
+    end
+
+    #
+    # Render a COMPACT_MANIFEST file in +#{staging_dir}/\+COMPACT_MANIFEST+
+    #
+    # @return [void]
+    #
+    def write_compact_manifest
+      destination = File.join(staging_dir, "+COMPACT_MANIFEST")
+      File.open(destination, "w+") do |f|
+        FFI_Yajl::Encoder.encode(compact_manifest, pretty: true).each_line do |line|
+          f.write(line)
+        end
+      end
+    end
+
+    #
+    # Render a MANIFEST file in +#{staging_dir}/\+MANIFEST+
+    #
+    # @return [void]
+    #
+    def write_manifest
+      destination = File.join(staging_dir, "+MANIFEST")
+      File.open(destination, "w+") do |f|
+        FFI_Yajl::Encoder.encode(compact_manifest.merge(manifest), pretty: true).each_line do |line|
+          f.write(line)
+        end
+      end
+    end
+
+    #
+    # Create the +.txz+ file, compressing at xz level 9. The use of the
+    # +fakeroot+ command is required so that the package is owned by
```

Comment:
Is this comment still accurate? I don't see any `fakeroot` invocation there... But I could be missing something.